### PR TITLE
add workflow client with tests

### PR DESCRIPTION
**Pull request by jreidinger**

### CODE REVIEW

File: src/lib/migration/main_workflow.rb [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+      },
+      "restore"      => {
+        abort: :abort
+      }
+    }
+
+    def aliases
+      {
+        "restore"      => ->() { restore_state },
+        "repositories" => -> { repositories }
```
Comment (lslezak) [inline src/lib/migration/main_workflow.rb]: Inconsistency found, this should be rather `->()` as on the line above.

Comment (jreidinger) [inline src/lib/migration/main_workflow.rb]: interesting that it works, something is strange. maybe we do need it at all?

File: src/lib/migration/main_workflow.rb [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+# To contact SUSE about this file by physical or electronic mail, you may find
+# current contact information at www.suse.com.
+# ------------------------------------------------------------------------------
+
+require "yast"
+
+Yast.import "Sequencer"
+
+module Migration
+  # Goal of class is to provide main single entry point to start migration
```
Comment (lslezak) [inline src/lib/migration/main_workflow.rb]: NP: `The goal of the class ...`

File: src/lib/migration/main_workflow.rb (hunk 1 above)
Comment (lslezak) [inline src/lib/migration/main_workflow.rb]: Um, may be we do not need it (should be investigated later), but we should use one form consitently (IIRC `->()` form is already used).

Comment (lslezak) [inline src/lib/migration/main_workflow.rb]: Maybe rubocop suggested using `->()` ?

Comment (jreidinger) [inline src/lib/migration/main_workflow.rb]: rubocop do not complain, interesting :) otherwise it won't pass travis check

**Comment by lslezak** [comment]
LGTM

**Pull request merged**